src/store/AuthStore.js

```diff
@@ -0,0 +1,54 @@
+import { types } from 'mobx-state-tree'
+import oauth from 'panoptes-client/lib/oauth'
+
+import { ASYNC_STATES } from '@util'
+
+const AuthStore = types.model('AuthStore', {
+  
+  status: types.optional(types.string, ASYNC_STATES.IDLE),
+  user: types.optional(types.frozen({}), null),  // When uninitialised, user should be null instead of {}
+  
+}).actions(self => {
+  return {
+    
+    initialise () {
+      self.checkUser()
+    },
+    
+    checkUser () {
+      self.setStatus(ASYNC_STATES.FETCHING)
+      oauth.checkCurrent()
+      .then(user => {
+        self.setStatus(ASYNC_STATES.SUCCESS)
+        self.setUser(user)
+      })
+    },
+    
+    login () {
+      oauth.signIn(computeRedirectURL(window))
+    },
+    
+    logout () {
+      oauth.signOut()
+      .then(user => {
+        self.setUser(user)
+      })
+    },
+    
+    setStatus (val) {
```

Comment:
I think MST model properties are getters/setters by default, so you don't have to define your own setters. Just do `self.status = 'some value'` or `self.user = user`.

Reply:
Noted - these changes will be made in #23 since I made some big updates to `.setStatus` that'll make rebasing a headache if I changed things here.